### ISSUE TITLE
What's the idea behind adding max_box_per_image as an input to YOLO ?

### ISSUE THREAD

**Issue by andohuman** [experiencor/basic-yolo-keras#195]
I pretty much understand the rest of the code to an extent but can't wrap my head around this. Some detailed explanation would be nice. Thanks in advance.

**Comment by experiencor** [comment]
@andohuman It's the max number of bounding boxes per images, which depends on the dataset.

**Comment by andohuman** [comment]
@experiencor does adding that as an input increase the performance ? I don't get why that is an input to the model.

**Comment by experiencor** [comment]
@andohuman It's used as the dimension prior of true_boxes, which is a convenient way to simplify the loss function.

**Comment by sadakmed** [comment]
@experiencor  do you mean that max_box_per_image is the N of objects possible that can be detected in an image. so if max_box_per_image was 1  it can only detect one object in the image, although maybe two objects or more?
